src/SourceBuild/content/test/Microsoft.DotNet.Tests/PackageSourceMappingsTests.cs

```diff
@@ -0,0 +1,242 @@
+        [Theory]
+        [InlineData("sb-sbrp-offline.config", false)]
+        [InlineData("sb-sbrp-online.config", true)]
+        public void SourceBuildSbrpRepoTests(string nugetConfigFilename, bool useOnlineFeeds)
+        {
+            string[] sources = ["prebuilt", "previously-source-built", "reference-packages"];
+            RunTest(nugetConfigFilename, useOnlineFeeds, sources);
+        }
+
+        private static void RunTest(string nugetConfigFilename, bool useOnlineFeeds, string[] sources, string[]? customSources = null, bool sourceBuild = true)
+        {
+            string psmAssetsDir = Path.Combine(Directory.GetCurrentDirectory(), "assets", "PackageSourceMappingsTests");
+            string originalNugetConfig = Path.Combine(psmAssetsDir, "original", nugetConfigFilename);
+            string expectedNugetConfig = Path.Combine(psmAssetsDir, "expected", nugetConfigFilename);
+
+            string modifiedNugetConfig = Path.Combine(PackageSourceMappingsSetup.PackageSourceMappingsRoot, nugetConfigFilename);
+            Directory.CreateDirectory(Path.GetDirectoryName(modifiedNugetConfig)!);
+            File.Copy(originalNugetConfig, modifiedNugetConfig, true);
+            UpdateNugetConfigTokens(modifiedNugetConfig);
+
+            var task = new UpdateNuGetConfigPackageSourcesMappings()
+            {
+                SbrpCacheSourceName = "source-build-reference-package-cache",
+                SbrpRepoSrcPath = TestSetup.SourceBuildReferencePackagesRepoDir,
+                SourceBuiltSourceNamePrefix = "source-built-",
+                NuGetConfigFile = modifiedNugetConfig,
+                BuildWithOnlineFeeds = useOnlineFeeds,
+                SourceBuildSources = sources,
+                CustomSources = customSources
+            };
+
+            if (sourceBuild)
+            {
+                task.ReferencePackagesSourceName = "reference-packages";
+                task.PreviouslySourceBuiltSourceName = "previously-source-built";
+                task.PrebuiltSourceName = "prebuilt";
+            }
+
+            task.Execute();
+
+            TokenizeLocalNugetConfigFeeds(modifiedNugetConfig);
+
+            string expectedNugetConfigContents = File.ReadAllText(expectedNugetConfig);
+            string modifiedNugetConfigContents = File.ReadAllText(modifiedNugetConfig);
+            Assert.Equal(expectedNugetConfigContents, modifiedNugetConfigContents);
+        }
+
+        private static void UpdateNugetConfigTokens(string nugetConfigFile)
+        {
+            string fileContents = File.ReadAllText(nugetConfigFile);
+            foreach (KeyValuePair<string, string> kvp in TestSetup.LocalTokenSourceMappings)
+            {
+                fileContents = fileContents.Replace(kvp.Key, kvp.Value);
+            }
+            File.WriteAllText(nugetConfigFile, fileContents);
+        }
+
+        private static void TokenizeLocalNugetConfigFeeds(string nugetConfigFile)
```

Comment:
What is the different with this method versus UpdateNugetConfigTokens?  The implementations appear identical.

Reply:
One is replacing `key` with `value`, the other replacing `value` with `key`.

This was a result of refactoring, but I stopped too soon. I will refactor this further.

Reply:
Fixed with https://github.com/dotnet/sdk/pull/44649/commits/f13f4f1cfb5a7adec023571eaf49ddcc8a051648